src/phoenix/server/api/routers/v1/spans.py

```diff
@@ -128,3 +134,149 @@ async def content() -> AsyncIterator[bytes]:
+      404:
+        description: Span not found
+    """
+    payload: List[Dict[str, Any]] = await request.json()
+    span_gids = [GlobalID.from_id(annotation["span_id"]) for annotation in payload]
+
+    resolved_span_ids = []
+    for span_gid in span_gids:
+        try:
+            resolved_span_ids.append(from_global_id_with_expected_type(span_gid, "Span"))
+        except ValueError:
+            return Response(
+                content="Span with ID {span_gid} does not exist",
+                status_code=HTTP_404_NOT_FOUND,
+            )
+
+    async with request.app.state.db() as session:
+        spans = await session.execute(
+            select(models.Span).filter(models.Span.id.in_(resolved_span_ids))
+        )
+        existing_span_ids = {span.id for span in spans.scalars()}
+
+        missing_span_ids = set(resolved_span_ids) - existing_span_ids
+        if missing_span_ids:
+            missing_span_gids = [
+                str(GlobalID("Span", str(span_gid))) for span_gid in missing_span_ids
+            ]
+            return Response(
+                content=f"Spans with IDs {', '.join(missing_span_gids)} do not exist.",
+                status_code=HTTP_404_NOT_FOUND,
+            )
+
+        inserted_annotations = []
+        for annotation in payload:
+            span_gid = GlobalID.from_id(annotation["span_id"])
+            span_id = from_global_id_with_expected_type(span_gid, "Span")
+            name = annotation["name"]
+            annotator_kind = annotation["annotator_kind"]
+            result = annotation.get("result")
+            label = result.get("label") if result else None
+            score = result.get("score") if result else None
+            explanation = result.get("explanation") if result else None
+            error = annotation.get("error")
+            metadata = annotation.get("metadata") or {}
+
+            values = dict(
+                span_rowid=span_id,
+                name=name,
+                label=label,
+                score=score,
+                explanation=explanation,
+                error=error,
+                annotator_kind=annotator_kind,
+                metadata_=metadata,
+            )
+            set_ = {
+                **{k: v for k, v in values.items() if k != "metadata_"},
+                "metadata": values["metadata_"],
+            }
+
+            dialect = SupportedSQLDialect(session.bind.dialect.name)
+            span_annotation = await session.scalar(
+                insert_on_conflict(
+                    dialect=dialect,
+                    table=models.SpanAnnotation,
+                    values=values,
+                    constraint="uq_span_annotations_span_rowid_name",
+                    column_names=("span_rowid", "name"),
+                    on_conflict=OnConflict.DO_UPDATE,
+                    set_=set_,
+                ).returning(models.SpanAnnotation)
```

Comment:
```suggestion
                ).returning(models.SpanAnnotation.id)
```

Looks like we only need the id since that is all that is being returned.